presto-main/src/main/java/com/facebook/presto/operator/HttpRpcShuffleClient.java

```diff
@@ -65,17 +67,31 @@
 
     private final HttpClient httpClient;
     private final URI location;
+    private final Optional<URI> asyncPageTransportLocation;
 
     public HttpRpcShuffleClient(HttpClient httpClient, URI location)
+    {
+        this(httpClient, location, false);
+    }
+
+    public HttpRpcShuffleClient(HttpClient httpClient, URI location, boolean asyncPageTransportEnabled)
     {
         this.httpClient = requireNonNull(httpClient, "httpClient is null");
         this.location = requireNonNull(location, "location is null");
+        if (asyncPageTransportEnabled) {
+            String path = location.getPath().replace("v1/task", "v2/task");
```

Comment:
This seems a bit hacky... though modifying the location from the source could be complicated and overkill.  Is there a way to switch on this via perhaps a query param?